### PR TITLE
Bump the installed version of rust to 1.79.0

### PR DESCRIPTION
**Pull request by jad-zd**
# Description

This PR bumps the rust toolchain's version from "nightly-2023-12-11" to the latest version of the stable channel "1.79.0" and the multiversx-sc crate from "0.45.1" to its latest version "0.50.5".

The reason for this is that multiversx-sc version >= "0.50.5" requires, version 1.78.0 and above of the rust toolchain (which is newer than "nightly-2023-12-11").

## Changes

- A user can now specify which version of the rust toolchain he wants to be installed, by default version 1.79.0 is installed (changing the install target `wasm32-unknown-unknown` is still not possible, as this is the only target for the MultiversX Blockchain): changes in `xsuite/src/cli/installRustCmd.ts` and `xsuite/src/cli/helpers.ts` ;
- Smart contracts have been rebuilt with these new versions: changes in `contracts/` and `xsuite/contracts` ;
- 7 tests have been updated to match the new code hashes of the rebuilt smart contracts: changes in `xsuite/src/data/index.test.ts`, `xsuite/src/world/fsworld.test.ts` and `xsuite/src/world/lsworld.test.ts` ;
- The integration test that simulates a user installation, build and tests has been adapted as it uses the smart contracts of the main branch which use version 0.45.1 of multiversx-sc and require an older `rustc` version ( :exclamation: Note: this change must be reverted after the next release :exclamation: ): changes in `xsuite/src/cli/cli.test.ts`.

### CODE REVIEW

File: xsuite/src/cli/cli.test.ts [hunk 1]
```diff
@@ -347,12 +347,23 @@ test.concurrent(
   async () => {
     using c = newContext();
 
+    // Switch to a rust version compatible with sample contracts on main branch.
```
Comment (lcswillems) [inline xsuite/src/cli/cli.test.ts:350]: Add a comment here to mention it should be removed

File: xsuite/src/cli/cli.test.ts [hunk 2]
```diff
@@ -426,6 +437,9 @@ test.concurrent(
       chalk.cyan(`$ ${binaryPath} .`),
       "",
     ]);
+
+    // Switch back to the default rust version.
+    await c.cmd("install-rust");
```
Comment (lcswillems) [inline xsuite/src/cli/cli.test.ts:442]: Add a comment here to mention it should be removed eventually

File: xsuite/src/cli/installRustCmd.ts [hunk 3]
```diff
@@ -12,12 +12,17 @@ export const addInstallRustCmd = (cmd: Command) => {
     .description(
       `Install Rust with rustup: toolchain ${rustToolchain} & target ${rustTarget}.`,
     )
+    .option(
```
Comment (lcswillems) [inline xsuite/src/cli/installRustCmd.ts:15]: I think we could set a default value here? And this way `toolchain` would then be always defined

File: xsuite/src/cli/installRustCmd.ts [hunk 4]
```diff
@@ -12,12 +12,17 @@ export const addInstallRustCmd = (cmd: Command) => {
...
+      "--toolchain <TOOLCHAIN>",
+      `Rust toolchain version, defaults to ${rustToolchain}`,
+    )
     .action(action);
 };
 
-const action = () => {
+const action = ({ toolchain }: { toolchain?: string }) => {
+  const toolchain2Install = toolchain ? toolchain : rustToolchain;
```
Comment (lcswillems) [inline xsuite/src/cli/installRustCmd.ts:23]: `toolchain2Install` is not correct English. 

Maybe renaming `rustToolchain` into `defaultRustToolchain` and then:

```
toolchain ??= defaultRustToolchain
```

?

**Review by lcswillems — CHANGES_REQUESTED** [review]

File: xsuite/src/cli/cli.test.ts (hunk 2 above)
Comment (jad-zd) [reply xsuite/src/cli/cli.test.ts:444]: Done

File: xsuite/src/cli/installRustCmd.ts (hunk 3 above)
Comment (jad-zd) [reply xsuite/src/cli/installRustCmd.ts:15]: Done

File: xsuite/src/cli/installRustCmd.ts (hunk 4 above)
Comment (jad-zd) [reply xsuite/src/cli/installRustCmd.ts]: Done

File: xsuite/src/cli/installRustCmd.ts (hunk 3 above)
Comment (lcswillems) [reply xsuite/src/cli/installRustCmd.ts:15]: We should rather use:

https://github.com/tj/commander.js?tab=readme-ov-file#default-option-value

And line 23 we should not need to provide the default value

**Review by lcswillems — CHANGES_REQUESTED** [review]

**Pull request merged**